oo/classical-inheritance-es5.md

```diff
@@ -0,0 +1,166 @@
+Many programming languages use the concepts of `class`, `subclass` and `inheritance` to assist with code reuse. When we create objects of a particular subclass these objects inherit properties and methods from their parent class. This allows us to write code once on the parent class and have it inherited and reused on its descendants. 
+
+JavaScript ES5 does not implement "classical inheritance" as cleanly as other languages but its common to mimic classical inheritance with both constructor functions and factory functions in ES5 codebases. As these are not true classes we sometimes call these "pseudo-classes". The examples below show how to implement classical inheritance with the tools available in ES5:
+
+## Classical inheritance with constructor functions
+
+```js
+function Post (options) {
+  this.author = options.author
+  this.content = options.content
+  this.createdAt = Date.now()
+}
+
+Post.prototype.getCreatedAt = function () {
+  return this.createdAt
+}
+
+function Comment () {
+  Post.prototype.constructor.apply(this, arguments) // call the "parent" pseudo-class constructor 
```

Comment:
I can understand why you'd write it this way for illustration purposes, but it might be worth mentioning that you can shorten it to `Post.apply(this, arguments)`.